main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+)
+
+var wg sync.WaitGroup
```

Comment:
There appear to be three ways of using a `sync.WaitGroup`:

1. Use a Package Level Variable - as in this code

1. Use a Clojure
   ```go
   wg.Add(1)
   go func() {
       // do your thing  
       wg.Done()
   }()
   wg.Wait()
   ```

1. Pass a Pointer to the Function
   ```go
   func startURLCheckers(n int, c chan string, wg *sync.WaitGroup) {
   	defer wg.Done()
   	// do your thing
   }
   ```

The general view is that a clojure seems to be OK for a small number of lines but much beyond that and you'd be better of passing a pointer into your functions.